File: scratch/clean_mock_db.py
import os
import re

controllers_dir = r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\backend\controllers"

# Regex to match the if block:
# if (global.MOCK_DB || ...) { ... }
# Note: Since there can be nested braces, it's better to just do a simple line-based approach or use a recursive descent parser.
# Actually, since the formatting is consistent, we can just look for `if (global.MOCK_DB` and find the matching closing brace.

def remove_mock_db_blocks(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # We will search for "if (global.MOCK_DB"
    while True:
        start_idx = content.find("if (global.MOCK_DB")
        if start_idx == -1:
            break
        
        # Find the opening brace of this if block
        brace_start = content.find("{", start_idx)
        if brace_start == -1:
            break
            
        # Find the matching closing brace
        brace_count = 1
        idx = brace_start + 1
        while brace_count > 0 and idx < len(content):
            if content[idx] == '{':
                brace_count += 1
            elif content[idx] == '}':
                brace_count -= 1
            idx += 1
            
        if brace_count == 0:
            # We found the end of the block.
            # Remove from start_idx up to idx
            # Also try to remove leading spaces before 'if'
            line_start = content.rfind('\n', 0, start_idx)
            if line_start != -1:
                content = content[:line_start+1] + content[idx:]
            else:
                content = content[:start_idx] + content[idx:]
        else:
            break

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

for filename in os.listdir(controllers_dir):
    if filename.endswith(".js"):
        filepath = os.path.join(controllers_dir, filename)
        remove_mock_db_blocks(filepath)
        print(f"Processed {filename}")

File: scratch/clean_mock_db.js
const fs = require('fs');
const path = require('path');

const controllersDir = path.join(__dirname, '..', 'backend', 'controllers');

function removeMockDbBlocks(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');

    while (true) {
        let startIdx = content.indexOf('if (global.MOCK_DB');
        if (startIdx === -1) break;

        let braceStart = content.indexOf('{', startIdx);
        if (braceStart === -1) break;

        let braceCount = 1;
        let idx = braceStart + 1;
        while (braceCount > 0 && idx < content.length) {
            if (content[idx] === '{') braceCount++;
            else if (content[idx] === '}') braceCount--;
            idx++;
        }

        if (braceCount === 0) {
            let lineStart = content.lastIndexOf('\n', startIdx);
            if (lineStart !== -1) {
                content = content.substring(0, lineStart + 1) + content.substring(idx);
            } else {
                content = content.substring(0, startIdx) + content.substring(idx);
            }
        } else {
            break;
        }
    }

    fs.writeFileSync(filepath, content, 'utf8');
}

fs.readdirSync(controllersDir).forEach(file => {
    if (file.endsWith('.js')) {
        removeMockDbBlocks(path.join(controllersDir, file));
        console.log(`Processed ${file}`);
    }
});
